.github folder maintaince

john-westcott-iv opened the pull request:
<!--- changelog-entry
# Fill in 'msg' below to have an entry automatically added to the next release changelog.
# Leaving 'msg' blank will not generate a changelog entry for this PR.
# Please ensure this is a simple (and readable) one-line string.
---
msg: ""
-->

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Cleaning and modifying the .github folder for the following:
  * Removing old awxbot files
      * BOTMETA
      * ISSUE_TEMPLATE
      * LABEL_MAP
  * Added top level config file for new issue display.
  * Modified bug report:
      * Changed heading wording (to tell the difference between a bug and a feature)
      * Added Collection, CLI and Other component options
      * Added N/A to required awx install method (for docs/cli/etc components)
      * Removed "docker on *" and boot2docker options in favor of "docker development environment"
      * Moved technical support information to top level config file
   * Modified feature requests template:
      * Converted from .md to .yml file
      * Changed heading wording (to tell the difference between a bug and a feature)
      * Added terms (to match bugs)
      * Added components (to match bugs)
   * Removed security_ bug_issue template as github now shows our SECURITY.md as a policy on issue creation.
   * Modified Pull Request template:
      * Changed issue types from [feature, bugs, docs] to [breaking, feature, bug] (to match X, Y, Z releases)
      * Aligned component name with issues/bug forms
   * Modified issue labeler:
      * Cleaned up types for new wording of headers
      * Added components: awx_collection and cli
   * Modified PR labeler:
      * Changed from component:collection to awx_collection
      * Added "dependencies" tag for specific files
   * Modified stage workflow (and PR issue template) to remove change log generation
   * Removed CODEOWNERS

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - GitHub

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 21.0.1.dev191+gaa24b377eb.d20220606
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

john-westcott-iv committed:
Removing old awxbot files

john-westcott-iv committed:
Removing security bug report as GitHub now shows the security piolicy from /SECURITY.md

john-westcott-iv committed:
Changing feature_request from md to yml

john-westcott-iv committed:
Adding additional options to bug report components andinstall method

john-westcott-iv committed:
Removing old ISSUE_TEMPLATE.md

john-westcott-iv committed:
Changing issue type and adding additional components

john-westcott-iv committed:
Removing auto-generated change log

john-westcott-iv committed:
Adding awx_collection and cli components

Changing language for type labels

john-westcott-iv committed:
Changing from collection to awx_collection tag and adding dependencies tag

john-westcott-iv committed:
Adding unicode bug to bug repot to match feature unicode character

john-westcott-iv committed:
Changing bug to bug or docs

gundalow commented:
AFAIK you can't test GitHub Issue forms without making them "live", so you could update https://github.com/john-westcott-iv/awx to make `pr_changes` the default branch (and enable issues)

john-westcott-iv commented:
> AFAIK you can't test GitHub Issue forms without making them "live", so you could update https://github.com/john-westcott-iv/awx to make `pr_changes` the default branch (and enable issues)

I actually made a dummy repo https://github.com/john-westcott-iv/github-file-test/issues/new/choose which is just the SECURITY.md and the .github directory so I can see the forms there. I haven't tried submitting anything to validate that the labels work nor creating a PR but we could test all of that if we think these changes are a good idea.

gundalow commented:
Do folks see any value in copying any more of https://github.com/ansible/ansible/issues/new/choose



> > AFAIK you can't test GitHub Issue forms without making them "live", so you could update https://github.com/john-westcott-iv/awx to make `pr_changes` the default branch (and enable issues)
> 
> I actually made a dummy repo https://github.com/john-westcott-iv/github-file-test/issues/new/choose which is just the SECURITY.md and the .github directory so I can see the forms there. I haven't tried submitting anything to validate that the labels work nor creating a PR but we could test all of that if we think these changes are a good idea.

Nice use of a temp repo :)

gundalow commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -1,11 +1,11 @@
       value: |
-        Issues are for **concrete, actionable bugs and feature requests** only. For debugging help or technical support, please use:
+        Bug Report issues are for **concrete, actionable bugs** only. For debugging help or technical support, please use:
         - The #ansible-awx channel on irc.libera.chat
         - The awx project mailing list, https://groups.google.com/forum/#!forum/awx-project
```
We can put markdown in here.

```suggestion
        - The [awx:ansible.com](https://matrix.to/#/#awx:ansible.com) channel on Matrix
        - The `#ansible-awx` channel on irc.libera.chat
        - The [AWX project mailing list](https://groups.google.com/forum/#!forum/awx-project)
```

john-westcott-iv replied:
We might also want to consider trying to reference https://github.com/ansible/awx/#get-involved so that we have a single place for links to the mailing list/IRC/etc.

gundalow commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,43 @@
+    attributes:
+      value: |
+        Feature Request issues are for **feature requests** only. For debugging help or technical support, please use:
+        - The #ansible-awx channel on irc.libera.chat
```
Same update as .github/ISSUE_TEMPLATE/bug_report.yml

john-westcott-iv replied:
Another option might be to move these message to the "top level" inside of a config.yml file so the IRC/Google group is right at the top of the new issue screen.

john-westcott-iv replied:
We changed this wording and added md style links.

john-westcott-iv committed:
Adding changes per merge meeting

Use CLI instead of awxkit

Remove docker on * and boot2docker infavor of docker development environmnet

Create top level issue with: CoC, Enterprise, Top level help

Remove old CODEOWNERS file

john-westcott-iv committed:
Minor changes from testing
